--- clang/lib/Driver/ToolChains/RISCVToolchain.cpp	2022-07-14 14:39:14.918208804 +0000
+++ clang/lib/Driver/ToolChains/RISCVToolchain.cpp	2022-07-14 14:39:14.918208804 +0000
@@ -98,7 +98,7 @@
   if (!D.SysRoot.empty())
     CmdArgs.push_back(Args.MakeArgString("--sysroot=" + D.SysRoot));
 
-  std::string Linker = getToolChain().GetProgramPath(getShortName());
+  std::string Linker = getToolChain().GetLinkerPath();
 
   bool WantCRTs =
       !Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles);
